prov/sm2/src/sm2.h

```diff
@@ -123,6 +129,36 @@ struct sm2_xfer_entry {
 	uint8_t user_data[SM2_INJECT_SIZE];
 } __attribute__((packed));
 
+struct sm2_atomic_secondary_hdr {
```

Comment:
Is there an sm2_atomic_primary_hdr? If not, I would just rename to sm2_atomic_hdr